Turn exercise 005 into practice

MissJae opened the pull request:
This Pr is in reference to task #2170.
I am still a little confused about the task description, I would appreciate feedback on my attempt if possible @cuihtlauac 

Fix PR #2170

Commit:
Turn exercise 005 into practice

cuihtlauac reviewed (changes requested):
Thanks @MissJae, this is great.

This PR shouln't not touch folder 003, remove all the changes touching this folder from this PR.

Your work on folder 005 is good. It compiles and passes the tests. Here are the changes which are also needed:

- Add newline at the end of each file which don't
- Add an additional test case with `rev []` (call it "nil")

cuihtlauac commented:
Was commit c1d45de70d7939391ad4e04a4e6e0081bca145c9 a mistake?

cuihtlauac reviewed (changes requested):
Rename folder `005.` into `005`. Add newlines at end of file. Do not use git push with force option

Commit:
Fixed Pr requested cahnges.
Added a new line at the end of each file.
Added an additional test case with rev [] and called it "nil".

MissJae committed:
Merge branch 'ocaml:main' into Draft--Turn-exercise-005-into-practice

MissJae commented:
> Was commit [c1d45de](https://github.com/ocaml/ocaml.org/commit/c1d45de70d7939391ad4e04a4e6e0081bca145c9) a mistake?

Sorry it was a mistake. I have removed it

cuihtlauac commented on practice/005./answer/impl.ml:
```diff
@@ -0,0 +1,7 @@
+    | h :: t -> aux (h :: acc) t
+  in
+  aux [] list
+  
```
```suggestion
```

Remove white space and add newline at end of file

Commit:
Rename folder from 005. to 005

MissJae committed:
Merge branch 'ocaml:main' into Draft--Turn-exercise-005-into-practice

MissJae commented:
@cuihtlauac please can you check this? I've made the requested changes.

cuihtlauac commented on practice/005/answer/impl.ml:
```diff
@@ -0,0 +1,7 @@
+    | h :: t -> aux (h :: acc) t
+  in
+  aux [] list
+  
```
```suggestion
  aux [] list

```

cuihtlauac committed:
Update practice/005/answer/impl.ml

cuihtlauac reviewed (approved):
Thanks @MissJae

divyankachaudhari commented:
@MissJae @cuihtlauac This doesn't compile.

```
~/ocaml.org/practice/005 (main ✘)✭ ᐅ dune build --root .                              
File "ex.ml", line 9, characters 42-57:
9 |     "nil" >:: (fun _ -> assert_equal None (Tested.rev []));
                                              ^^^^^^^^^^^^^^^
Error: This expression has type 'a list
       but an expression was expected of type 'b option
 ```

MissJae commented:
> @MissJae @cuihtlauac This doesn't compile.
> 
> ```
> ~/ocaml.org/practice/005 (main ✘)✭ ᐅ dune build --root .                              
> File "ex.ml", line 9, characters 42-57:
> 9 |     "nil" >:: (fun _ -> assert_equal None (Tested.rev []));
>                                               ^^^^^^^^^^^^^^^
> Error: This expression has type 'a list
>        but an expression was expected of type 'b option
> ```

Hey @divyankachaudhari I think I've figured out the error. I think I need to change None to []. @cuihtlauac should I create a pull request for it?
